src/runtime/virtcontainers/kata_agent.go

```diff
@@ -797,6 +789,13 @@ func (k *kataAgent) startSandbox(ctx context.Context, sandbox *Sandbox) error {
 		kmodules = setupKernelModules(k.kmodules)
 	}
 
+	// If a Policy has been specified, send it to the agent.
```

Comment:
I vote for keeping the SetPolicy call above the updateInterfaces, updateRoutes, and addARPNeighbors calls. That wouldn't result in a lot of duplication and would maintain the ability to possibly enforce Policy rules for the updateInterfaces, updateRoutes, and addARPNeighbors calls in the non-remote hypervisor case.

Reply:
Sure, I can move it above the if block instead of below it

Reply:
I decided to keep the separate remote hypervisor context just in case it takes a little longer to set the policy. It's added an extra if, but it's not too bad. Please take a look Dan and see if this works okay for you. Thanks!